### PR TITLE
KDOC

### PR DESCRIPTION
**Pull request by octa-one**

### CODE REVIEW

**Comment by ilslv9** [comment]
lgtm

File: alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/reduce/LinkedNavReducer.kt [hunk 1]
```diff
@@ -2,10 +2,28 @@ package com.github.octaone.alcubierre.reduce
 
 import com.github.octaone.alcubierre.annotation.AlcubierreInternalApi
 
+/**
+ * Linked list implementation of [NavReducer].
+ * The reduce starts at the head and traverses to the tail.
+ *
+ * Each reducer has access to the head to have an ability ti restart the entire chain.
```
Comment (Batonec) [inline alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/reduce/LinkedNavReducer.kt:9]: To restart

File: alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/reduce/LinkedNavReducer.kt [hunk 2]
```diff
@@ -2,10 +2,28 @@ package com.github.octaone.alcubierre.reduce
...
+ * As an example, there's a chain:
+ * HeadReducer -> ForwardReducer -> DeeplinkReducer
+ * DeeplinkReducer waits for an action with the deeplink, processes it and triggers a chain of reducers with Forward action.
+ * But after DeeplinkReducer, there are no more reducers for Forward reducing.
```
Comment (Batonec) [inline alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/reduce/LinkedNavReducer.kt:13]: But there are no more reducers in the downstream, so, Forward reducing is impossible.

File: alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/reduce/NavReducer.kt [hunk 3]
```diff
@@ -2,7 +2,18 @@ package com.github.octaone.alcubierre.reduce
 
 import com.github.octaone.alcubierre.action.AnyNavAction
 
+/**
+ * Interface for state update logic.
+ */
 public interface NavReducer<S> {
 
+    /**
+     * The reducer function that specifies how the state gets updated based on [action].
```
Comment (Batonec) [inline alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/reduce/NavReducer.kt:11]: "Gets updated" - that sounds a little bit sketchy

File: alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/reduce/builder/ReducerLinkedListBuilder.kt [hunk 4]
```diff
@@ -7,17 +7,27 @@ import com.github.octaone.alcubierre.annotation.AlcubierreInternalApi
 import com.github.octaone.alcubierre.reduce.LinkedNavReducer
 import com.github.octaone.alcubierre.reduce.NavReducer
 
+/**
+ * Reducers chain (linked list) builder.
+ * Allows to add new Reducers without the need of manually linking them to the added ones.
```
Comment (Batonec) [inline alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/reduce/builder/ReducerLinkedListBuilder.kt:12]: Allows adding without the need for linking

File: alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/screen/Dialog.kt [hunk 5]
```diff
@@ -7,7 +7,14 @@ import com.github.octaone.alcubierre.screen.extra.ExtrasContainer
 import java.util.UUID
 
 /**
- * Base dialog
+ * Abstraction for describing dialog destination for navigation.
+ *
+ * The main difference from [Screen] is that the [Dialog] is shown outside the stack on top of any content.
+ * And also there can only be one shown dialog at a time.
```
Comment (Batonec) [inline alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/screen/Dialog.kt:13]: Furthermore, only one dialog can be displayed at a time.

File: alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/screen/Dialog.kt [hunk 6]
```diff
@@ -7,7 +7,14 @@ import com.github.octaone.alcubierre.screen.extra.ExtrasContainer
...
+ * The remaining added dialogs will be queued according to [priority].
```
Comment (Batonec) [inline alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/screen/Dialog.kt:14]: Are gonna be)

Comment (Batonec) [reply alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/screen/Dialog.kt:14]: And I'd simplify that

File: alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/screen/Screen.kt [hunk 7]
```diff
@@ -6,7 +6,14 @@ import com.github.octaone.alcubierre.screen.extra.ExtrasContainer
...
 /**
- * Base screen (specific destination for navigation)
+ * Abstraction for describing a destination for navigation.
+ *
+ * Be aware:
+ * The library only uses random [screenId] to compare screens.
+ * This makes possible to add multiple identical screens to the backstack.
+ *
+ * Be careful when using the `object` modifier for screens.
+ * Adding such object to the backstack multiple times may cause errors.
```
Comment (Batonec) [inline alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/screen/Screen.kt:16]: Might

File: alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/screen/Screen.kt [hunk 8]
```diff
@@ -6,7 +6,14 @@ import com.github.octaone.alcubierre.screen.extra.ExtrasContainer
 import java.util.UUID
 
 /**
- * Base screen (specific destination for navigation)
+ * Abstraction for describing a destination for navigation.
+ *
+ * Be aware:
+ * The library only uses random [screenId] to compare screens.
+ * This makes possible to add multiple identical screens to the backstack.
```
Comment (Batonec) [inline alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/screen/Screen.kt:13]: Makes it possible?

File: alcubierre-condition/src/main/kotlin/com/github/octaone/alcubierre/condition/ConditionalTarget.kt [hunk 9]
```diff
@@ -1,9 +1,46 @@
...
 
+import com.github.octaone.alcubierre.condition.action.resolveCondition
+import com.github.octaone.alcubierre.condition.reducer.ConditionReducer
 import com.github.octaone.alcubierre.screen.extra.ExtrasContainer
 import com.github.octaone.alcubierre.screen.extra.LazyExtrasContainer
 import kotlin.reflect.KClass
 
+/**
+ * A virtual target for conditional deeplink navigation.
+ * Sometimes it is not possible to annotate an specific screen with Deeplink annotation,
```
Comment (Batonec) [inline alcubierre-condition/src/main/kotlin/com/github/octaone/alcubierre/condition/ConditionalTarget.kt:11]: A specific screen

File: alcubierre-condition/src/main/kotlin/com/github/octaone/alcubierre/condition/ConditionalTarget.kt [hunk 10]
```diff
@@ -1,9 +1,46 @@
...
+ * because that screen might be disabled by a remote config or might be part of an A/B test.
```
Comment (Batonec) [inline alcubierre-condition/src/main/kotlin/com/github/octaone/alcubierre/condition/ConditionalTarget.kt:12]: A part

File: alcubierre-condition/src/main/kotlin/com/github/octaone/alcubierre/condition/NavCondition.kt [hunk 11]
```diff
@@ -1,14 +1,31 @@
...
 
+/**
+ * An interface for classes that encapsulate conditional navigation logic.
+ */
 public interface NavCondition {
 
-    public fun resolve(target: ConditionalTarget, state: AnyRootNavState): AnyNavAction
+    /**
+     * Condition resolving function.
+     * @param target [ConditionalTarget] on which this [NavCondition] was triggered.
```
Comment (Batonec) [inline alcubierre-condition/src/main/kotlin/com/github/octaone/alcubierre/condition/NavCondition.kt:14]: Has been?

File: alcubierre-condition/src/main/kotlin/com/github/octaone/alcubierre/condition/action/ResolveCondition.kt [hunk 12]
```diff
@@ -3,10 +3,19 @@ package com.github.octaone.alcubierre.condition.action
...
 
+/**
+ * Action to resolve a conditional target.
+ * @see ConditionalTarget
+ * @see ConditionReducer
+ */
 public data class ResolveCondition<S : Screen, D : Dialog>(val conditionalTarget: ConditionalTarget): NavAction<S, D>
 
+/**
+ * Extension to dispatch action [ResolveCondition].
```
Comment (Batonec) [inline alcubierre-condition/src/main/kotlin/com/github/octaone/alcubierre/condition/action/ResolveCondition.kt:18]: For dispatching

File: alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/reduce/NavReducer.kt (hunk 3 above)
Comment (octa-one) [reply alcubierre-base/src/main/kotlin/com/github/octaone/alcubierre/reduce/NavReducer.kt:11]: modified

**Pull request merged**